payments-core/src/main/java/com/stripe/android/payments/PaymentFlowResultProcessor.kt

```diff
@@ -175,7 +175,8 @@ internal sealed class PaymentFlowResultProcessor<T : StripeIntent, out S : Strip
      */
     private fun shouldCallRefreshIntent(stripeIntent: StripeIntent): Boolean {
         return stripeIntent.paymentMethod?.type == PaymentMethod.Type.WeChatPay ||
-            stripeIntent.paymentMethod?.type == PaymentMethod.Type.Upi
+            stripeIntent.paymentMethod?.type == PaymentMethod.Type.Upi ||
+            stripeIntent.paymentMethod?.type == PaymentMethod.Type.CashAppPay
```

Comment:
Nit: We could move this into an extension property of `PaymentMethod.Type` if this continues to grow.

Reply:
For sure!

I see us removing `shouldRefreshIfIntentRequiresAction` in favor of a sealed class where we define the behavior.

Maybe called something like `AfterRedirectAction` - poll, refresh, none

Right now our polling behavior has a fixed retry count, but some LPMs want a shorter duration, so we could provide some input into the constructor for that behavior.